specs/Menu.md

```diff
@@ -1,731 +1,545 @@
 
-```TSX
-function() {
-    return (
-    <Slots.Root {...props} role={"menuitem"}>
-        {props.children}
-    </Slots.Root>);
+```typescript
+const menuProps = {
+  items: [{
+    ...
+    data: { foo: "bar" }
+  }]
 }
 ```
 
-#### Submenu
+v0 custom rendering through shorthand components is a consistent experience through all shorthand components, but provide a smaller API surface (whether this is simpler or less powerful can be subjective). Custom rendering in the case of the `Menu` component would be done through the use of `chilren` prop either through the standard React child component API or through shorthand as a callback function.
+
+```typescript
+// v0 shorthand children render callback
+const items = [
+  {
+    key: 'editorials',
+    children: (El, props) => <El>{props.key}</El>
+  },
+]
+
+<Menu defaultActiveIndex={0} items={items} />
+
+// v0 children API custom render
+<Menu>
+  <Menu.Item index={0}>
+    <Menu.ItemContent>Editorials</Menu.ItemContent>
+  </Menu.Item>
+  <Menu.Item index={1}>
+    CustomContent
+  </Menu.Item>
+  {/*Not recommended but definitely possible*/}
+  <div>custom item</div>
+</Menu>
+```
 
-Maybe should be named SubmenuContext?
-Should submenu have its own context?
+## Variants
 
-```TSX
-function() {
-return (
-    <OpenableContext.Provider value={submenuState}>
-        {props.children}
-    </OpenableContext.Provider>);
-}
-```
+### Nested menus
 
-#### SubmenuList
+A `Menu` should be able to trigger an additional instance of itself as a part of one or more of its options. The nested `Menu` component should have the same functional capabilities as the root `Menu` component.
 
-```TSX
-function() {
-    const openableContext = React.useContext(OpenableContext);
-    return (openableContext.open &&
-    <Slots.Root {...menuProps}>
-        {props.children}
-    </Slots.Root>);
-}
-```
+The actions that trigger the the nested `Menu` should be be consistent with the actions that can trigger any root `Menu` from a similar UI control.
 
-#### SubmenuItem
+We advise that no more than two nested `Menu` components be used, but this spec does not functionally apply that constrain to the implementation of the `Menu` component.
 
-```TSX
-function() {
-    const openableContext = React.useContext(OpenableContext);
-    return (
-    <Slots.Root {...menuItemProps} onClick={openableContext.toggleOpen}>
-        {props.children}
-    </Slots.Root>);
-}
+### Selection state
+
+A `Menu` should be able to track and represent the selection state of all or some of its options if required.
+
+When an options is associated with a selection state. The `Menu`, either root or nested, should control its dismiss behavior accordingly based on configuration.
+
+### Sections
+
+A `Menu` can be partitioned into sections using visible dividers in its list of options. Each section can contain a heading title that announces or briefly describes the options in the particular section
```

Comment:
for accessibility, wrapping sections in a `role='group'` might be necessary to give them a common title. But this needs to be verified.

Reply:
Yeah i would like run a11y attributes with milan and incorporate that here soon, it's already tracked in the checklist in description